solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -1588,20 +1590,32 @@ private CoreDescriptor reloadCoreDescriptor(CoreDescriptor oldDesc) {
     return ret;
   }
 
+  /**
+   * reloads a core
+   * refer {@link CoreContainer#reload(String, UUID)} for details
+   */
+  public void reload(String name) {
+    reload(name, null);
+  }
   /**
    * Recreates a SolrCore.
    * While the new core is loading, requests will continue to be dispatched to
    * and processed by the old core
    *
    * @param name the name of the SolrCore to reload
+   * @param coreId The unique Id of the core
```

Comment:
plus mention it's optional, and then if it's specified and the coreId doesn't match, then the reload is a no-op.  Maybe rename coreId to ifCoreIdMatches to reflect not only what it is but how it's used.